storage/internal/benchmarks/main.go

```diff
@@ -249,28 +251,53 @@ func main() {
 	benchGroup, _ := errgroup.WithContext(ctx)
 	benchGroup.SetLimit(opts.numWorkers)
 
+	exitWithErrorCode := false
+
 	// Run benchmarks
 	for i := 0; i < opts.numSamples && time.Since(start) < opts.timeout; i++ {
 		benchGroup.Go(func() error {
 			benchmark := w1r3{opts: opts, bucketName: opts.bucket}
-			if err := benchmark.setup(); err != nil {
-				log.Fatalf("run setup failed: %v", err)
+			if err := benchmark.setup(ctx); err != nil {
+				// If setup failed once, it will probably continue failing.
+				// Returning the error here will cancel the context to stop the
+				// benchmarking.
+				return fmt.Errorf("run setup failed: %v", err)
 			}
 			if err := benchmark.run(ctx); err != nil {
-				log.Fatalf("run failed: %v", err)
+				// If a run fails, we continue, as it could be a temporary issue.
+				// We log the error and make sure the program exits with an error
+				// to indicate that we did see an error, even though we continue.
+				log.Printf("run failed: %v", err)
+				exitWithErrorCode = true
+			}
+			if err := benchmark.cleanup(); err != nil {
+				// If cleanup fails, we continue, as a single fail is not critical.
+				// We log the error and make sure the program exits with an error
+				// to indicate that we did see an error, even though we continue.
+				// Cleanup may be expected to fail if there is an issue with the run.
+				log.Printf("run cleanup failed: %v", err)
+				exitWithErrorCode = true
 			}
 			return nil
 		})
 	}
 
-	benchGroup.Wait()
+	err := benchGroup.Wait()
```

Comment:
`Errgroup.Wait` seems to return a single error. Does it make sense to exit with just this error on L295? I guess other errors will be in the logs?

Reply:
I think it makes sense. Run and cleanup errors will be in the logs. Setup errors are the only ones that would cause `Errgroup.Wait` to fail. Since the context is cancelled when the first setup error is seen, it should be the first error and the one we want to print. Additional errors would happen after the context is cancelled, at which point they are no longer that relevant.